File: 560-subarray-sum-equals-k/subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        HashMap<Integer,Integer>mp=new HashMap<>();
        int total=0;
        int presum=0;
        mp.put(0,1);
        for(int i=0;i<nums.length;i++){
            presum+=nums[i];
            if(mp.containsKey(presum-k)){
                total+=mp.get(presum-k);
            }
            mp.put(presum,mp.getOrDefault(presum,0)+1);
        }
        return total;
    }
}

File: 560-subarray-sum-equals-k/subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int,int>mp;
        int presum=0;
        int total=0;
        mp[0]=1;
        for(int i=0;i<nums.size();i++){
            presum+=nums[i];
            if(mp.count(presum-k)){
                total+=mp[presum-k];
                mp[presum]++;
            }
            else{
                mp[presum]++;
            }
        }
        return total;
    }
};